src/model.rs

```diff
@@ -675,17 +721,30 @@ impl Model<ProblemCreated> {
         cons
     }
 
-    pub fn add_cons_coef(
-        &mut self,
-        cons: Rc<Constraint>,
-        var: Rc<Variable>,
-        coef: f64,
-    ) {
+    pub fn add_cons_set_part(&mut self, vars: Vec<Rc<Variable>>, name: &str) -> Rc<Constraint> {
+        assert!(vars.iter().all(|v| v.get_type() == VarType::Binary));
+        let cons = self
+            .scip
+            .create_cons_set_part(vars, name)
+            .expect("Failed to add constraint set partition in state ProblemCreated");
+        let cons = Rc::new(cons);
+        self.state.conss.push(cons.clone());
```

Comment:
why is a clone needed here, shouldn't it be a move or something, or is it because we are also returning the constraint